location.go

```diff
@@ -0,0 +1,168 @@
+/*
+management-interface - Web based management of Raspberry Pis over WiFi
+Copyright (C) 2018, The Cacophony Project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package managementinterface
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+const deviceLocationFile = "/etc/cacophony/location.yaml"
+
+// LocationHandler shows the location of the device.  The location can be viewed and/or set manually.
+func LocationHandler(w http.ResponseWriter, r *http.Request) {
+	type locationResponse struct {
+		Location     *rawLocationData
+		Message      string
+		ErrorMessage string
+	}
+
+	switch r.Method {
+	case "GET", "":
+		location, err := readLocationFile(deviceLocationFile)
+		resp := &locationResponse{
+			Location:     location.rawLocationData(),
+			ErrorMessage: errorMessage(err),
+		}
+		tmpl.ExecuteTemplate(w, "location.html", resp)
+
+	case "POST":
+		rawLocation, err := handleLocationPostRequest(w, r)
+		resp := &locationResponse{
+			Location:     rawLocation,
+			Message:      successMessage(err, "Location updated"),
+			ErrorMessage: errorMessage(err),
+		}
+		tmpl.ExecuteTemplate(w, "location.html", resp)
+
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+
+// APILocationHandler writes the location of the device to the deviceLocationFile
+func APILocationHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	_, err := handleLocationPostRequest(w, r)
+	if isClientError(err) {
+		w.WriteHeader(http.StatusBadRequest)
+	} else if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func handleLocationPostRequest(w http.ResponseWriter, r *http.Request) (*rawLocationData, error) {
```

Comment:
Does this need a doc string?

Reply:
Not really, it's not exported and the name is pretty clear.